ocaml/alerts/expiry_alert.ml

```diff
@@ -0,0 +1,167 @@
+(*
+ * Copyright (C) Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+module XenAPI = Client.Client
+module Date = Xapi_stdext_date.Date
+
+type message_name_t = string
+
+type message_priority_t = int64
+
+type message_id_t = message_name_t * message_priority_t
+
+type remaining_days_t = int
+
+type expiry_messaging_info_t = {
+    message_cls:
+      [ `Certificate
+      | `Host
+      | `PVS_proxy
+      | `Pool
+      | `SR
+      | `VDI
+      | `VM
+      | `VMPP
+      | `VMSS ]
+  ; message_obj_uuid: string
+  ; obj_description: string
+  ; message_sent_on_remaining_days_list: (remaining_days_t * message_id_t) list
```

Comment:
How about `priorities`, or `urgency_levels`?

Reply:
Do you mean change `remaining_days_t` to `priorities`? I had considered that way. But to let the lib generate the alert for certificates and CSS, a `expiry` datetime is still needed to generate the `message_body`, so I think it will not make it easier and clearer.

Reply:
I meant to rename the `message_sent_on_remaining_days_list`. It's too long.